bitnami/keycloak print connection refused while tring to connect to bitnami/postgresql

### Name and Version

bitnami/keycloak:20.0.5-debian-11-r8

### What architecture are you using?

amd64

### What steps will reproduce the bug?

Here is the `yaml` file and required `.env` file

```cmd
# Keycloak's envs 
KEYCLOAK_ADMIN_USER=admin
KEYCLOAK_ADMIN_PASSWORD=admin

# Keycloak db's envs
POSTGRESQL_DB=keycloak
POSTGRESQL_USER=postgres
POSTGRESQL_PASSWORD=admin
```


```yml
version: "3"

services:
  pgadmin4:
    image: dpage/pgadmin4:6.21
    ports:
      - 9000:80
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@admin.com
      PGADMIN_DEFAULT_PASSWORD: ${POSTGRESQL_PASSWORD}

  postgres:
    image: bitnami/postgresql:15.2.0-debian-11-r13
    restart: always
    environment:
      POSTGRESQL_PASSWORD: ${POSTGRESQL_PASSWORD}
      POSTGRESQL_USER: ${POSTGRESQL_USER}
      POSTGRESQL_DB: ${POSTGRESQL_DB}
    volumes:
      - keycloak_db_volume:/var/lib/postgresql/data

  keycloak:
    image: bitnami/keycloak:20.0.5-debian-11-r8
    restart: always
    depends_on:
      - postgres
    ports:
      - 8080:8080
      # - 8443:8443 HTTPS port
    volumes:
      - keycloak_volume:/tmp
    environment:
      KEYCLOAK_CREATE_ADMIN_USER: true
      KEYCLOAK_ADMIN_USER: ${KEYCLOAK_ADMIN_USER}
      KEYCLOAK_ADMIN_PASSWORD: ${KEYCLOAK_ADMIN_PASSWORD}
      KEYCLOAK_DATABASE_HOST: postgres
      KEYCLOAK_DATABASE_PORT: 5432
      KEYCLOAK_DATABASE_NAME: ${POSTGRESQL_DB}
      KEYCLOAK_DATABASE_USER: ${POSTGRESQL_USER}
      KEYCLOAK_DATABASE_PASSWORD: ${POSTGRESQL_PASSWORD}
      KEYCLOAK_DATABASE_SCHEMA: public
      KEYCLOAK_DATABASE_VENDOR: postgres
      # KEYCLOAK_ENABLE_STATISTICS: ${KEYCLOAK_ENABLE_STATISTICS}
      # KEYCLOAK_PRODUCTION: true In prod env and configure TLS pls

volumes:
  keycloak_db_volume:
  keycloak_volume:
```

then just do `docker compose up`

### What is the expected behavior?

Connect to db as I can do with pg admin with entering `postgres` as the host and other info as they are listed in the `.env` file.

### What do you see instead?

```cmd
you-say-keycloak-1  | keycloak 18:27:43.48 INFO  ==> ** Starting keycloak **
you-say-keycloak-1  | 2023-03-21 18:27:45,770 INFO  [org.keycloak.quarkus.runtime.hostname.DefaultHostnameProvider] (main) Hostname settings: Base URL: <unset>, Hostname: <request>, Strict HTTPS: false, Path: <request>, Strict BackChannel: false, Admin URL: <unset>, Admin: <request>, Port: -1, Proxied: true
you-say-keycloak-1  | 2023-03-21 18:27:47,526 WARN  [io.quarkus.agroal.runtime.DataSources] (main) Datasource <default> enables XA but transaction recovery is not enabled. Please enable transaction recovery by setting quarkus.transaction-manager.enable-recovery=true, otherwise data may be lost if the application is terminated abruptly
you-say-keycloak-1  | 2023-03-21 18:27:47,882 WARN  [io.agroal.pool] (agroal-11) Datasource '<default>': Connection to localhost:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
you-say-keycloak-1  | 2023-03-21 18:27:47,883 WARN  [org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator] (JPA Startup Thread: keycloak-default) HHH000342: Could not obtain connection to query metadata: org.postgresql.util.PSQLException: Connection to localhost:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
you-say-keycloak-1  |   at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:319)
you-say-keycloak-1  |   at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:49)
you-say-keycloak-1  |   at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:247)
you-say-keycloak-1  |   at org.postgresql.Driver.makeConnection(Driver.java:434)
you-say-keycloak-1  |   at org.postgresql.Driver.connect(Driver.java:291)
you-say-keycloak-1  |   at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:681)
you-say-keycloak-1  |   at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:229)
you-say-keycloak-1  |   at org.postgresql.ds.common.BaseDataSource.getConnection(BaseDataSource.java:103)
you-say-keycloak-1  |   at org.postgresql.xa.PGXADataSource.getXAConnection(PGXADataSource.java:49)
you-say-keycloak-1  |   at org.postgresql.xa.PGXADataSource.getXAConnection(PGXADataSource.java:35)
you-say-keycloak-1  |   at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
you-say-keycloak-1  |   at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:535)
you-say-keycloak-1  |   at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:516)
you-say-keycloak-1  |   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
you-say-keycloak-1  |   at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
you-say-keycloak-1  |   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
you-say-keycloak-1  |   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
you-say-keycloak-1  |   at java.base/java.lang.Thread.run(Thread.java:833)
you-say-keycloak-1  | Caused by: java.net.ConnectException: Connection refused
you-say-keycloak-1  |   at java.base/sun.nio.ch.Net.pollConnect(Native Method)
you-say-keycloak-1  |   at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
you-say-keycloak-1  |   at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:542)
you-say-keycloak-1  |   at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:597)
you-say-keycloak-1  |   at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
you-say-keycloak-1  |   at java.base/java.net.Socket.connect(Socket.java:633)
you-say-keycloak-1  |   at org.postgresql.core.PGStream.createSocket(PGStream.java:241)
you-say-keycloak-1  |   at org.postgresql.core.PGStream.<init>(PGStream.java:98)
you-say-keycloak-1  |   at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:109)
you-say-keycloak-1  |   at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:235)
you-say-keycloak-1  |   ... 17 more
you-say-keycloak-1  | 
you-say-keycloak-1  | 2023-03-21 18:27:48,642 WARN  [org.infinispan.PERSISTENCE] (keycloak-cache-init) ISPN000554: jboss-marshalling is deprecated and planned for removal
you-say-keycloak-1  | 2023-03-21 18:27:48,712 WARN  [org.infinispan.CONFIG] (keycloak-cache-init) ISPN000569: Unable to persist Infinispan internal caches as no global state enabled
you-say-keycloak-1  | 2023-03-21 18:27:48,767 INFO  [org.infinispan.CONTAINER] (keycloak-cache-init) ISPN000556: Starting user marshaller 'org.infinispan.jboss.marshalling.core.JBossUserMarshaller'
you-say-keycloak-1  | 2023-03-21 18:27:49,056 INFO  [org.infinispan.CONTAINER] (keycloak-cache-init) ISPN000128: Infinispan version: Infinispan 'Triskaidekaphobia' 13.0.10.Final
you-say-keycloak-1  | 2023-03-21 18:27:49,156 INFO  [org.infinispan.CLUSTER] (keycloak-cache-init) ISPN000078: Starting JGroups channel `ISPN`
you-say-keycloak-1  | 2023-03-21 18:27:49,156 INFO  [org.infinispan.CLUSTER] (keycloak-cache-init) ISPN000088: Unable to use any JGroups configuration mechanisms provided in properties {}. Using default JGroups configuration!
you-say-keycloak-1  | 2023-03-21 18:27:49,259 WARN  [org.jgroups.protocols.UDP] (keycloak-cache-init) JGRP000015: the send buffer of socket MulticastSocket was set to 1.00MB, but the OS only allocated 212.99KB
you-say-keycloak-1  | 2023-03-21 18:27:49,259 WARN  [org.jgroups.protocols.UDP] (keycloak-cache-init) JGRP000015: the receive buffer of socket MulticastSocket was set to 20.00MB, but the OS only allocated 212.99KB
you-say-keycloak-1  | 2023-03-21 18:27:49,260 WARN  [org.jgroups.protocols.UDP] (keycloak-cache-init) JGRP000015: the send buffer of socket MulticastSocket was set to 1.00MB, but the OS only allocated 212.99KB
you-say-keycloak-1  | 2023-03-21 18:27:49,261 WARN  [org.jgroups.protocols.UDP] (keycloak-cache-init) JGRP000015: the receive buffer of socket MulticastSocket was set to 25.00MB, but the OS only allocated 212.99KB
you-say-keycloak-1  | 2023-03-21 18:27:51,296 INFO  [org.jgroups.protocols.pbcast.GMS] (keycloak-cache-init) 4eb77f9f0696-38651: no members discovered after 2003 ms: creating cluster as coordinator
```

### Additional information

I am frustrated and kinda not know why the `KEYCLOAK_DATABASE_HOST` does not work and it says it trys to connect to `localhost:5432`. I also tried to publish 5432 port in postgres service but still it cannot connect to the database.

--- a/bitnami/keycloak/README.md
+++ b/bitnami/keycloak/README.md
@@ -85,7 +85,7 @@ The Bitnami Keycloak container can create a default admin user by setting the fo
 
 The Bitnami Keycloak container can connect to a database by setting the following environment variables:
 
-* `KEYCLOAK_DATABASE_VENDOR`: Database vendor. Default: **postgres**. Use **dev-mem** or **dev-file** to use an in-memory or file-based database for development purposes.
+* `KEYCLOAK_DATABASE_VENDOR`: Database vendor. Default: **postgresql**. Use **dev-mem** or **dev-file** to use an in-memory or file-based database for development purposes.
 
 ### PostgreSQL database connection configuration
 

